.drone.yml

```diff
@@ -89,13 +441,129 @@ steps:
 - name: docker-publish
   image: plugins/docker:windows-1809-amd64
   settings:
-    context: 1.14/windows/windowsservercore-1809
+    context: 1.12/windows/windowsservercore-1809
     custom_dns: 1.1.1.1
-    dockerfile: 1.14/windows/windowsservercore-1809/Dockerfile
+    dockerfile: 1.12/windows/windowsservercore-1809/Dockerfile
     password:
       from_secret: docker_password
     repo: rancher/golang
-    tag: 1.14-windowsservercore-1809
+    tag: 1.12-windowsservercore-1809
+    username:
+      from_secret: docker_username
+  volumes:
+  - name: docker_pipe
+    path: \\\\.\\pipe\\docker_engine
+  when:
+    event:
+    - push
+    instance:
+    - drone-publish.rancher.io
+    ref:
+    - refs/heads/master
+
+volumes:
+- name: docker_pipe
+  host:
+    path: \\\\.\\pipe\\docker_engine
+
+---
+kind: pipeline
+name: windows-go112-1903
+
+platform:
+  os: windows
+  arch: amd64
+  version: 1903
+
+steps:
+- name: docker-ci
+  image: plugins/docker:windows-1903-amd64
+  settings:
+    dry_run: true
+    context: 1.12/windows/windowsservercore-1903
+    custom_dns: 1.1.1.1
+    dockerfile: 1.12/windows/windowsservercore-1903/Dockerfile
+    password:
+      from_secret: docker_password
+    repo: rancher/golang
+    tag: 1.12-windowsservercore-1903
+    username:
+      from_secret: docker_username
+  volumes:
+  - name: docker_pipe
+    path: \\\\.\\pipe\\docker_engine
+  when:
+    event:
+    - pull_request
+
+- name: docker-publish
+  image: plugins/docker:windows-1903-amd64
+  settings:
+    context: 1.12/windows/windowsservercore-1903
+    custom_dns: 1.1.1.1
+    dockerfile: 1.12/windows/windowsservercore-1903/Dockerfile
+    password:
+      from_secret: docker_password
+    repo: rancher/golang
+    tag: 1.12-windowsservercore-1903
+    username:
+      from_secret: docker_username
+  volumes:
+  - name: docker_pipe
+    path: \\\\.\\pipe\\docker_engine
+  when:
+    event:
+    - push
+    instance:
+    - drone-publish.rancher.io
+    ref:
+    - refs/heads/master
+
+volumes:
+- name: docker_pipe
+  host:
+    path: \\\\.\\pipe\\docker_engine
+
+---
+kind: pipeline
+name: windows-go112-1909
+
+platform:
+  os: windows
+  arch: amd64
+  version: 1909
+
+steps:
+- name: docker-ci
+  image: plugins/docker:windows-1909-amd64
+  settings:
+    dry_run: true
+    context: 1.12/windows/windowsservercore-1909
+    custom_dns: 1.1.1.1
+    dockerfile: 1.12/windows/windowsservercore-1909/Dockerfile
+    password:
+      from_secret: docker_password
+    repo: rancher/golang
+    tag: 1.12-windowsservercore-1909
+    username:
+      from_secret: docker_username
+  volumes:
+  - name: docker_pipe
+    path: \\\\.\\pipe\\docker_engine
+  when:
+    event:
+    - pull_request
+
+- name: docker-publish
+  image: plugins/docker:windows-1909-amd64
```

Comment:
why does this one use a specific tag but not the go 1.14 one?

Reply:
mistake, fixed